tests/single_node_test.go

```diff
@@ -0,0 +1,61 @@
+package tests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+)
+
+var _ = Describe("Single Node Topology", func() {
+	BeforeEach(func() {
+		strategy.SkipUnlessSingleReplicaTopologyMode()
+	})
+
+	AfterEach(func() {
+		strategy.RevertToOriginalSspCr()
+		waitUntilDeployed()
+	})
+
+	It("[test_id:???] Number of Template Validator replicas is not bigger then one", func() {
+		watch, err := StartWatch(sspListerWatcher)
+		Expect(err).ToNot(HaveOccurred())
+		defer watch.Stop()
+
+		var newValidatorReplicas int32 = 3
+		updateSsp(func(foundSsp *sspv1beta1.SSP) {
+			foundSsp.Spec.TemplateValidator.Replicas = &newValidatorReplicas
+		})
+
+		// Watch changes until above change
+		err = WatchChangesUntil(watch, func(updatedSsp *sspv1beta1.SSP) bool {
```

Comment:
You are missing the second `WatchChangesUntil` call. The reason for 2 calls is to ensure that the operator has seen the above change done by `updateSsp`.

First `WatchChangesUntil` will trigger before the operator sees the change, because `updatedSsp.Generation > updatedSsp.Status.ObservedGeneration` .
And the second `WatchChangesUntil` should watch until `updatedSsp.Generation == updatedSsp.Status.ObservedGeneration`. To be sure the the operator has seen the change.

Reply:
oh and there was another problem there. Is this better now?
by the way can we have a timing issue here? by going to wait for ">" after it is already "=="?

Reply:
This is good.

There should not be a timing issue. The watch is started at the beginning of the test, before any changes to the SSP CR. A watch is basically a channel that contains all changes to a resource in chronological order, as they happened on the server. So an event with ">" will be before "==", because SSP operator must first see ">" and then change it to "==".